mne/gui/_ieeg_locate_gui.py

```diff
@@ -1013,31 +1047,27 @@ def _key_press_event(self, event):
         if event.key() in (QtCore.Qt.Key_Up, QtCore.Qt.Key_Down,
                            QtCore.Qt.Key_Left, QtCore.Qt.Key_Right,
                            QtCore.Qt.Key_PageUp, QtCore.Qt.Key_PageDown):
+            ras = np.array(self._ras)
             if event.key() in (QtCore.Qt.Key_Up, QtCore.Qt.Key_Down):
-                self._ras[2] += 2 * (event.key() == QtCore.Qt.Key_Up) - 1
+                ras[2] += 2 * (event.key() == QtCore.Qt.Key_Up) - 1
             elif event.key() in (QtCore.Qt.Key_Left, QtCore.Qt.Key_Right):
-                self._ras[0] += 2 * (event.key() == QtCore.Qt.Key_Right) - 1
+                ras[0] += 2 * (event.key() == QtCore.Qt.Key_Right) - 1
             elif event.key() in (QtCore.Qt.Key_PageUp,
                                  QtCore.Qt.Key_PageDown):
-                self._ras[1] += 2 * (event.key() == QtCore.Qt.Key_PageUp) - 1
-            self._move_cursors_to_pos()
+                ras[1] += 2 * (event.key() == QtCore.Qt.Key_PageUp) - 1
+            self._set_ras(ras)
 
-    def _on_click(self, axis, event):
+    def _on_click(self, event, axis):
         """Move to view on MRI and CT on click."""
-        # Transform coordinates to figure coordinates
-        pos = self._get_click_pos(axis, event.x, event.y)
-        logger.info(f'Clicked axis {axis} at pos {pos}')
-
-        if axis is not None and pos is not None:
-            xyz = apply_trans(self._ras_vox_t, self._ras)
-            if axis == 0:
-                xyz[[1, 2]] = pos
-            elif axis == 1:
-                xyz[[0, 2]] = pos
-            elif axis == 2:
-                xyz[[0, 1]] = pos
-            self._ras = apply_trans(self._vox_ras_t, xyz)
-            self._move_cursors_to_pos()
+        if event.inaxes is self._figs[axis].axes[0]:
```

Comment:
Very nice, much more elegant, I didn't know about this one, good to know for the future